pennylane/_qubit_device.py

```diff
@@ -297,17 +297,13 @@ def execute(self, circuit, **kwargs):
         if self.shots is not None or any(isinstance(m, sample_type) for m in circuit.measurements):
```

Comment:
let's use the value instead of multiple func calls
```suggestion
        is_lightning = self._is_lightning_device()

        if self.shots is not None or any(isinstance(m, sample_type) for m in circuit.measurements):
```